vita3k/gui/src/game_context_menu.cpp

```diff
@@ -0,0 +1,167 @@
+        });
+
+        if (gui.title_id == game.title_id) {
+            LOG_INFO_IF(fs::remove_all(game_path / game.title_id), "Successfully deleted '{} [{}]'.", game.title_id, game.title);
+
+            if (!fs::exists(game_path / game.title_id)) {
+                gui.delete_game_icon = true;
+                gui.game_selector.games.erase(games_index);
+            } else
+                LOG_ERROR("Failed to delete '{} [{}]'.", game.title_id, game.title);
+        }
+    }
+}
+
+void game_context_menu(GuiState &gui, HostState &host, bool *selected) {
+    const fs::path save_data_path{ fs::path(host.pref_path) / "ux0/user/00/savedata" };
+    const fs::path shaderlog_path{ fs::path(host.base_path) / "shaderlog" };
+
+    for (const auto &game : gui.game_selector.games) {
+
+        if (gui.title_id == game.title_id) {
+
+            // Context Game Menu
+            if (ImGui::BeginPopupContextItem("#game_context_menu")) {
+                if (ImGui::MenuItem("Boot", game.title.c_str()))
+                    selected[0] = true;
+                if (ImGui::BeginMenu("Copy game info")) {
+                    if (ImGui::MenuItem("ID and Name")) {
+                        ImGui::LogToClipboard();
+                        ImGui::LogText("%s [%s]", game.title_id.c_str(), game.title.c_str());
+                        ImGui::LogFinish();
+                    }
+                    if (ImGui::MenuItem("ID")) {
+                        ImGui::LogToClipboard();
+                        ImGui::LogText("%s", game.title_id.c_str());
+                        ImGui::LogFinish();
+                    }
+                    if (ImGui::MenuItem("Name")) {
+                        ImGui::LogToClipboard();
+                        ImGui::LogText("%s", game.title.c_str());
+                        ImGui::LogFinish();
+                    }
+                    ImGui::EndMenu();
+                }
+                if (ImGui::BeginMenu("Remove")) {
+                    if (ImGui::MenuItem("Game"))
+                        gui.delete_game_popup = true;
+                    if (ImGui::MenuItem("Save Data"))
+                        gui.delete_savedata_popup = true;
+                    if (ImGui::MenuItem("Shader Log")) {
+                        if (fs::exists(shaderlog_path / game.title_id))
+                            fs::remove_all(shaderlog_path / game.title_id);
+                    }
+                    ImGui::EndMenu();
+                }
+                ImGui::EndPopup();
+            }
+
+            static const ImVec2 CONTEXT_BUTTON_SIZE = ImVec2(124, 0);
+
+            // Delete Game Popup
+            if (gui.delete_game_popup)
+                ImGui::OpenPopup("Delete Game");
+            ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
+            if (ImGui::BeginPopupModal("Delete Game", &gui.delete_game_popup, ImGuiWindowFlags_AlwaysAutoResize)) {
```

Comment:
My brain is confused...

`BeginPopupModal` returns false if `OpenPopup` is not called I assume? Wanted to be sure.

Reply:
Only works like that